### PR TITLE
RD-2335 Added deployment name column to nodes list widget

### PR DESCRIPTION
**Pull request by kubama**
The pre-existing Deployment column was renamed to Deployment ID and is hidden by default. As a result nothing has changed in widget appearance except that the Deployment column now shows deployment name instead of deployment ID.

System tests: https://jenkins.cloudify.co/view/UI/job/Stage-UI-System-Test/871

### CODE REVIEW

**Review by stepek — APPROVED** [review]
lgtm!

File: test/cypress/integration/widgets/nodes_spec.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+describe('Nodes list widget', () => {
```
Comment (Gelio) [inline test/cypress/integration/widgets/nodes_spec.ts:1]: :medal_military: for adding new tests for the widget :tada:

File: test/cypress/integration/widgets/nodes_spec.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    const deployment1Id = `${blueprintName}_dep1_id`;
+    const deployment1Name = `${blueprintName}_dep1_name`;
+    const deployment2Id = `${blueprintName}_dep2_id`;
+    const deployment2Name = `${blueprintName}_dep2_name`;
+
+    before(() =>
+        cy
+            .activate('valid_trial_license')
+            .usePageMock('nodes', { fieldsToShow: ['Deployment', 'Deployment ID'] })
+            .mockLogin()
```
Comment (Gelio) [inline test/cypress/integration/widgets/nodes_spec.ts:12]: Minor:

Would it not be safer/faster to `mockLogin` after all the blueprints are deployed?

File: test/cypress/integration/widgets/nodes_spec.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            .deleteDeployments(blueprintName, true)
```
Comment (Gelio) [inline test/cypress/integration/widgets/nodes_spec.ts:13]: Minor:

This looks to be optional, as `deleteBlueprints` will delete respective deployments too:

https://github.com/cloudify-cosmo/cloudify-stage/blob/0de621cbc844559d430ec97197e39be486ba49cd/test/cypress/support/blueprints.ts#L33-L35

We can keep it, though, for readability :smile:

File: test/cypress/integration/widgets/nodes_spec.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                { server_ip: '127.0.0.1' },
+                { display_name: deployment2Name }
+            )
+    );
+
+    it('should display nodes list', () => {
+        cy.setBlueprintContext(blueprintName);
+        cy.get('tbody tr').should('have.length', 2);
+        cy.get('table')
+            .getTable()
```
Comment (Gelio) [inline test/cypress/integration/widgets/nodes_spec.ts:34]: Nice, makes the test much more readable :+1:

File: test/cypress/integration/widgets/nodes_spec.ts [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+            .should(tableData => {
+                expect(tableData[0].Deployment).to.eq(deployment1Name);
+                expect(tableData[0]['Deployment ID']).to.eq(deployment1Id);
+                expect(tableData[1].Deployment).to.eq(deployment2Name);
+                expect(tableData[1]['Deployment ID']).to.eq(deployment2Id);
+            });
+
+        cy.setDeploymentContext(deployment1Id);
+        cy.contains('th', 'Deployment').should('not.exist');
```
Comment (Gelio) [inline test/cypress/integration/widgets/nodes_spec.ts:43]: Hmm... what does this test exactly? Should we not list that one single deployment with ID `deployment1Id` here?

Comment (Gelio) [reply test/cypress/integration/widgets/nodes_spec.ts:43]: Minor:

If we expect there are no rows, maybe let's assert that `No Results Found` or some other `noDataMessage` is displayed?

**Review by Gelio — APPROVED** [review]
Looks generally ok, I do have a few minor clarification questions :smile: I'll approve anyway, as those seem minor

File: test/cypress/integration/widgets/nodes_spec.ts (hunk 5 above)
Comment (kubama) [reply test/cypress/integration/widgets/nodes_spec.ts:43]: This assertion tests if 'Deployment' and 'Deployment ID' columns get hidden when deployment is set in the context - please note the `th` selector.

File: test/cypress/integration/widgets/nodes_spec.ts (hunk 3 above)
Comment (kubama) [reply test/cypress/integration/widgets/nodes_spec.ts:13]: Ah, yes, thanks for reminding though.

File: test/cypress/integration/widgets/nodes_spec.ts (hunk 2 above)
Comment (kubama) [reply test/cypress/integration/widgets/nodes_spec.ts:12]: I think it doesn't matter - the test body starts with setting blueprint context, which causes widget content to be reloaded, so it's perfectly safe. There would be also no speed benefit in switching the order.

Comment (Gelio) [reply test/cypress/integration/widgets/nodes_spec.ts:12]: Ok, makes sense

File: test/cypress/integration/widgets/nodes_spec.ts (hunk 5 above)
Comment (Gelio) [reply test/cypress/integration/widgets/nodes_spec.ts:43]: Right, makes sense :smile: So it tests the `data.deploymentSelected` part. And, conveniently, it also tests both columns. Clever, almost too clever :smile: 

Thanks for explaining

**Pull request merged**